server/helpers/cache.js

```diff
@@ -0,0 +1,38 @@
+const TIMEOUT = 24 * 60 * 60 * 1000 // 1 day
+const PRUNE_RATIO = 0.1
+
+class Cache {
+  constructor(limit, maxAge = TIMEOUT) {
+    this.limit = limit
+    this.maxAge = maxAge
+    this.cache = {}
+    this.size = 0
+  }
+
+  get(key) {
+    if (key in this.cache && this.cache[key].timestamp + this.maxAge > Date.now()) {
+      return this.cache[key].data
+    }
+    return undefined
+  }
+
+  set(key, value) {
+    if (this.get(key) === undefined) this.size++
+    this.cache[key] = {
+      timestamp: Date.now(),
+      data: value,
+    }
+    if (this.size > this.limit) this.prune()
+  }
+
+  prune() {
```

Comment:
I guess this method isn't really supposed to be called from outside so I'd underscore it (declaring it "private" would be ideal, but we don't have TS server side :/)

Reply:
Refactored